apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -638,43 +638,50 @@ static void utc_audio_pcm_readi_p(void)
 {
 	int fd;
 	char input_str[16];
-	ssize_t size;
-	ssize_t bytes;
-	int frames_read = 0;
-	unsigned int total_frames_read = 0;
-	unsigned int rate;
-	int remain;
 	int ret;
 	char *str;
 
-	size = pcm_get_buffer_size(g_pcm);
-	rate = pcm_get_rate(g_pcm);
-	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+	char *buffer;
+	unsigned int size;
+	unsigned int frames_read;
+	unsigned int total_frames_read;
+	unsigned int bytes_per_frame;
+	int capturing = 1;
 
-	g_record_buffer = (char *)malloc(bytes);	//rate * bit * channels * duration
-	TC_ASSERT_NEQ("pcm_readi_p", g_record_buffer, NULL);
+	size = pcm_frames_to_bytes(g_pcm, pcm_get_buffer_size(g_pcm));
+	buffer = malloc(size);
+	TC_ASSERT_NEQ("pcm_readi", buffer, NULL);
+
+	fd = open(AUDIO_TEST_FILE, O_RDWR | O_CREAT);
+	TC_ASSERT_GT("pcm_readi", fd, 0);
 
-	remain = pcm_bytes_to_frames(g_pcm, bytes);
-	printf("Record will be start for 3s, press any key to start(Total frame: %d, %d bytes)\n", remain, bytes);
+	printf("Record will be start for 3s, press any key to start\n");
 	fflush(stdout);
 	str = gets(input_str);
 	TC_ASSERT_NEQ_CLEANUP("pcm_readi", str, NULL, free(g_record_buffer));
```

Comment:
1. Where is allocation of "g_record_buffer"? Isn't it "buffer"?
2. Don't you need to close fd at failure case?

Reply:
buffer is right. I'll fix this.